### PR TITLE
Don't add extraRefs if the job belongs to the same org/repo with refs.

### PR DESCRIPTION
**Pull request by droslean**
If the rehearse job belongs to the same org/repo with the refs, the `extrarefs` injection is not needed. Otherwise, `clonerefs` will fail.

### CODE REVIEW

File: test/pj-rehearse-integration/expected.yaml [hunk 1]
```diff
@@ -512,11 +512,6 @@
         entrypoint: gcr.io/k8s-prow/entrypoint:v20190129-0a3c54c
         initupload: gcr.io/k8s-prow/initupload:v20190129-0a3c54c
         sidecar: gcr.io/k8s-prow/sidecar:v20190129-0a3c54c
-    extra_refs:
-    - base_ref: cluster-profile
-      org: openshift
-      repo: origin
-      workdir: true
```
Comment (petr-muller) [inline test/pj-rehearse-integration/expected.yaml]: This looks suspicious... This does not seem like the case where `extra_refs` should be removed (refs are not nil and are openshift/release, extra_refs are openshift/origin)

:confused:

File: pkg/rehearse/jobs.go [hunk 2]
```diff
@@ -101,12 +101,19 @@ func makeRehearsalPresubmit(source *prowconfig.Presubmit, repo string, prNumber
...
-			rehearsal.ExtraRefs = append(rehearsal.ExtraRefs, pjapi.Refs{
-				Org:     orgRepo[0],
-				Repo:    orgRepo[1],
-				BaseRef: branch,
-				WorkDir: true,
-			})
+			jobOrg := orgRepo[0]
+			jobRepo := orgRepo[1]
+
+			if refs != nil {
```
Comment (petr-muller) [inline pkg/rehearse/jobs.go]: I think we need to add `ExtraRefs` when `refs==nil` though, for periodics?

Comment (droslean) [reply pkg/rehearse/jobs.go]: there is no org/repo to add. In the periodics, the user has to add the `extra_refs` by himself.

File: test/pj-rehearse-integration/expected.yaml (hunk 1 above)
Comment (droslean) [reply test/pj-rehearse-integration/expected.yaml]: No, quite the opposite. This presubmit is under the `openshift/origin` repo in the [job config](https://github.com/openshift/ci-tools/blob/master/test/pj-rehearse-integration/master/ci-operator/jobs/super/duper/super-duper-cluster-profile-presubmits.yaml#L2).

Comment (petr-muller) [reply test/pj-rehearse-integration/expected.yaml]: Well, yeah - that's why I would expect it to have openshift/release refs (to report stuff to github) and openshift/origin extra_refs with workdir: true to be able to work with its code.

Comment (droslean) [reply test/pj-rehearse-integration/expected.yaml]: OH! yea

File: pkg/rehearse/jobs.go (hunk 2 above)
Comment (petr-muller) [reply pkg/rehearse/jobs.go]: Ah, we're relying on `repo==""` for periodics...

**Comment by petr-muller** [comment]
@droslean Would you mind adding one more testcase for https://github.com/openshift/ci-tools/pull/143#discussion_r325172349, e.g. having same org and different repo? It would be useful, we managed to hit that pitfall :)

**Comment by droslean** [comment]
Sure thing.

**Comment by droslean** [comment]
@petr-muller test case added.

**Pull request merged**